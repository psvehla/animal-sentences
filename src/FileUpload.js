import React, { Component } from 'react';
import axios from 'axios';
import './FileUpload.css';

class FileUpload extends Component {

  constructor(props) {
    super(props);
    this.state = {
    }
  }

  // checkMimeType=(event)=>{
  //
  //   let file = event.target.files[0];
  //
  //   if (file && file.type !== 'text/plain') {
  //     this.setState({
  //       output: "Only plain text files are supported.",
  //     })
  //
  //     return false;
  //   }
  //
  //   return true;
  // }
  //
  // checkFileSize=(event)=>{
  //
  //   let file = event.target.files[0];
  //
  //   if (file && file.size > 1048576 ) {
  //     this.setState({
  //       output: "Your file is too big, please try another one.",
  //     })
  //
  //     return false;
  //   }
  //
  //   return true;
  // }
  //
  // onChangeHandler=event=>{
  //   if (this.checkMimeType(event) && this.checkFileSize(event)) {
  //     this.setState({
  //       selectedFile: event.target.files[0],
  //       loaded: 0,
  //     })
  //   }
  // }

  // onClickHandler=()=>{
  //   const data = new FormData();
  //   data.append('file', this.state.selectedFile);
  //   data.append('prefixLen', this.state.prefixLen);
  //   data.append('suffixLen', this.state.suffixLen);
  //   data.append('numberOfSentences', this.state.numberOfSentences);
  //
  //   axios.post(process.env.REACT_APP_SENTENCE_SERVICE_ENDPOINT, data, {})
  //     .then(res => {
  //       this.setState({
  //         output: res.data,
  //       });
  //     })
  //     .catch((error) => {
  //       if (error.response) {
  //         this.setState({
  //           output: "Your file is invalid. Please upload a plain text one.",
  //         })
  //         console.log(error.response.data);
  //         console.log(error.response.status);
  //         console.log(error.response.headers);
  //       }
  //       else if (error.request) {
  //         console.log(error.request);
  //       }
  //       else {
  //         console.log('Error', error.message);
  //       }
  //       console.log(error.config);
  //     });
  // }

  render() {
    return (
      <div>
        <div className="form-group files">
          <label id="upload-label">Upload Your File </label>
          <input type="file" className="form-control" multiple={false} onChange={this.props.onChange}/>
        </div>
        <button type="button" className="btn btn-success btn-block" onClick={this.props.onClick}>Upload</button>
      </div>
    );
  }
}

export default FileUpload;
