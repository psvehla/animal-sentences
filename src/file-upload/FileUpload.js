import React, { Component } from 'react';
import './FileUpload.css';

class FileUpload extends Component {

  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    return (
      <div>
        <div className="form-group files">
          <label id="upload-label">Upload Your File </label>
          <input type="file" className="form-control" multiple={false} onChange={this.props.onChange}/>
        </div>
        <button type="button" className="btn btn-success btn-block" onClick={this.props.onClick}>Upload</button>
      </div>
    );
  }
}

export default FileUpload;
