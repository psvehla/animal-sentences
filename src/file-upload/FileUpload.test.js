import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FileUpload from './FileUpload';

it('renders correctly', () => {

  const tree = renderer
    .create(<FileUpload />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('check the onChange event', () => {

    const onChange = jest.fn(),
        props = {
            value: 10,
            onChange
        },
        component = mount(<FileUpload {...props} />).find('input');

    component.simulate('change', { target: {value: 20} });
    expect(onChange).toHaveBeenCalledTimes(1);
});

test('check the onClick event', () => {

    const onClick = jest.fn(),
        props = {
            onClick
        },
        component = mount(<FileUpload {...props} />).find('button');

    component.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
});
