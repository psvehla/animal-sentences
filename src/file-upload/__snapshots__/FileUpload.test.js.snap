// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="form-group files"
  >
    <label
      id="upload-label"
    >
      Upload Your File 
    </label>
    <input
      className="form-control"
      multiple={false}
      type="file"
    />
  </div>
  <button
    className="btn btn-success btn-block btn btn-primary"
    disabled={false}
    type="button"
  >
    Upload
  </button>
</div>
`;
